File: SDKs/interzoid.com/getglobalnumberinfo/1.0.0/python/src/sdk/models/operations/getglobalnumberinfo.py
from dataclasses import dataclass, field
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils


@dataclass
class GetglobalnumberinfoQueryParams:
    intlnumber: str = field(metadata={'query_param': { 'field_name': 'intlnumber', 'style': 'form', 'explode': True }})
    license: str = field(metadata={'query_param': { 'field_name': 'license', 'style': 'form', 'explode': True }})
    

@dataclass_json
@dataclass
class Getglobalnumberinfo200ApplicationJSON:
    code: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Code') }})
    country: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Country') }})
    credits: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Credits') }})
    language2: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Language2') }})
    language3: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Language3') }})
    mobile: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Mobile') }})
    primary_city: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('PrimaryCity') }})
    primary_language: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('PrimaryLanguage') }})
    region: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Region') }})
    wealth: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Wealth') }})
    

@dataclass
class GetglobalnumberinfoRequest:
    query_params: GetglobalnumberinfoQueryParams = field()
    

@dataclass
class GetglobalnumberinfoResponse:
    content_type: str = field()
    status_code: int = field()
    getglobalnumberinfo_200_application_json_object: Optional[Getglobalnumberinfo200ApplicationJSON] = field(default=None)
    

File: SDKs/interzoid.com/getglobalnumberinfo/1.0.0/typescript/src/sdk/models/operations/getglobalnumberinfo.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



export class GetglobalnumberinfoQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=intlnumber" })
  intlnumber: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=license" })
  license: string;
}


export class Getglobalnumberinfo200ApplicationJson extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=Code" })
  code?: string;

  @SpeakeasyMetadata({ data: "json, name=Country" })
  country?: string;

  @SpeakeasyMetadata({ data: "json, name=Credits" })
  credits?: string;

  @SpeakeasyMetadata({ data: "json, name=Language2" })
  language2?: string;

  @SpeakeasyMetadata({ data: "json, name=Language3" })
  language3?: string;

  @SpeakeasyMetadata({ data: "json, name=Mobile" })
  mobile?: string;

  @SpeakeasyMetadata({ data: "json, name=PrimaryCity" })
  primaryCity?: string;

  @SpeakeasyMetadata({ data: "json, name=PrimaryLanguage" })
  primaryLanguage?: string;

  @SpeakeasyMetadata({ data: "json, name=Region" })
  region?: string;

  @SpeakeasyMetadata({ data: "json, name=Wealth" })
  wealth?: string;
}


export class GetglobalnumberinfoRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: GetglobalnumberinfoQueryParams;
}


export class GetglobalnumberinfoResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  getglobalnumberinfo200ApplicationJsonObject?: Getglobalnumberinfo200ApplicationJson;
}
